incubator/exporter-stdout/src/main/java/io/aklivity/zilla/runtime/exporter/stdout/internal/printer/PrintableEventsStream.java

```diff
@@ -0,0 +1,154 @@
+    private final MqttEventFW mqttEventRO = new MqttEventFW();
+    private final SchemaRegistryEventFW schemaRegistryEventRO = new SchemaRegistryEventFW();
+    private final TcpEventFW tcpEventRO = new TcpEventFW();
+    private final TcpRemoteAccessFailedEventFW tcpRemoteAccessFailedEventRO = new TcpRemoteAccessFailedEventFW();
+    private final TlsEventFW tlsEventRO = new TlsEventFW();
+
+    private final LabelReader labels;
+    private final EventsLayoutReader layout;
+    private final PrintStream out;
+    private final Int2ObjectHashMap<MessageConsumer> eventHandlers;
+
+    public PrintableEventsStream(
+        LabelReader labels,
+        EventsLayoutReader layout,
+        PrintStream out)
+    {
+        this.labels = labels;
+        this.layout = layout;
+        this.out = out;
+
+        final Int2ObjectHashMap<MessageConsumer> eventHandlers = new Int2ObjectHashMap<>();
+        addEventHandler(labels, eventHandlers, "tcp", this::handleTcpEvent);
+        addEventHandler(labels, eventHandlers, "http", this::handleHttpEvent);
+        // TODO: Ati - add more
+        this.eventHandlers = eventHandlers;
+    }
+
+    private void addEventHandler(
+        LabelReader labels,
+        Int2ObjectHashMap<MessageConsumer> eventHandlers,
+        String type,
+        MessageConsumer consumer)
+    {
+        int labelId = labels.lookupLabelId(type);
+        if (labelId != 0)
+        {
+            eventHandlers.put(labelId, consumer);
+        }
+    }
+
+    public int process()
+    {
+        return layout.eventsBuffer().spy(this::handleEvent, 1);
+    }
+
+    private boolean handleEvent(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        final MessageConsumer handler = eventHandlers.get(msgTypeId);
+        if (handler != null)
+        {
+            handler.accept(msgTypeId, buffer, index, length);
+        }
+        return true;
+    }
+
+    private void handleTcpEvent(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        final TcpEventFW event = tcpEventRO.wrap(buffer, index, index + length);
+        switch (event.kind())
+        {
+        case REMOTE_ACCESS_FAILED:
+            TcpRemoteAccessFailedEventFW e = event.remoteAccessFailed();
+            out.printf(TCP_REMOTE_ACCESS_FAILED_FORMAT, e.timestamp(), e.traceId(), asBinding(e.bindingId()),
+                asString(e.address()));
+            break;
+        }
+    }
+
+    private void handleHttpEvent(
+        int msgTypeId,
+        DirectBuffer buffer,
+        int index,
+        int length)
+    {
+        // TODO: Ati
+    }
+
+    private String asBinding(
+        long bindingId)
+    {
+        String namespace = labels.lookupLabel(namespaceId(bindingId));
+        String binding = labels.lookupLabel(localId(bindingId));
+        return String.format("%s.%s", namespace, binding);
```

Comment:
Use `EngineContext.supplyNamespace(namespacedId)` and `EngineContext.supplyLocalName(namespacedId)`.
Avoid concatenating the parts of the binding name into a temporary string before applying them to the format string.